pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -285,7 +286,7 @@ public synchronized void readMoreEntries() {
                 }
 
                 havePendingReplayRead = true;
-                minReplayedPosition = messagesToReplayNow.stream().min(PositionImpl::compareTo).orElse(null);
+                minReplayedPosition = messagesToReplayNow.first();
```

Comment:
You need to check the `messagesToReplayNow.isEmpty()`.

Reply:
thanks for review, `messagesToReplayNow.isEmpty()` is called in line 282